Test/FileStorage/ControllerFunctionsSpec.hs

```diff
@@ -0,0 +1,60 @@
+module Test.FileStorage.ControllerFunctionsSpec where
+
+import Test.Hspec
+import IHP.Prelude
+import IHP.FileStorage.ControllerFunctions
+import IHP.Controller.Context
+import IHP.FrameworkConfig
+import IHP.ModelSupport
+import IHP.FileStorage.Types
+import System.IO.Temp (withSystemTempDirectory)
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.Text as Text
+import Data.Default (def)
+import Data.Time.Clock (getCurrentTime, addUTCTime)
+import Network.Wai as Wai (defaultRequest)
+import Network.Wai.Parse (FileInfo(..))
+import IHP.Controller.RequestContext
+
+tests :: Spec
+tests = describe "IHP.FileStorage.ControllerFunctions" $ do
+    describe "storeFileWithOptions" $ do
+        it "returns the objectPath without the baseUrl" $ do
+            withSystemTempDirectory "ihp-test" $ \tempDir -> do
+                context <- createControllerContext
+                let ?context = context
+
+                let fileInfo = FileInfo
+                        { fileName = "test.txt"
+                        , fileContentType = "text/plain"
+                        , fileContent = "Hello, world!"
+                        }
+
+                result <- storeFile fileInfo "Test.FileStorage.ControllerFunctionsSpec"
+
+                result.url `shouldBe` "Test.FileStorage.ControllerFunctionsSpec/test.txt"
+
+    describe "createTemporaryDownloadUrlFromPathWithExpiredAt" $ do
+        it "returns baseUrl concatenated with objectPath when objectPath does not start with http:// or https://" $ do
+            context <- createControllerContext
+            let ?context = context
+            let objectPath = "static/test.txt"
+            temporaryDownloadUrl <- createTemporaryDownloadUrlFromPath objectPath
+
+            temporaryDownloadUrl.url `shouldBe` "http://localhost:8000/static/test.txt"
+
+        it "returns '/' concatenated with objectPath when objectPath starts with 'http://' or 'https://'" $ do
+            context <- createControllerContext
+            let ?context = context
+            let objectPath = "https://example.com/static/test.txt"
+            temporaryDownloadUrl <- createTemporaryDownloadUrlFromPath objectPath
+
+            temporaryDownloadUrl.url `shouldBe` "https://example.com/static/test.txt"
+
+createControllerContext = do
+    let
+        requestBody = FormBody { params = [], files = [] }
+        request = Wai.defaultRequest
+        requestContext = RequestContext { request, respond = error "respond", requestBody, frameworkConfig = error "frameworkConfig" }
```

Comment:
@mpscholten :wave: I suspect this line, with the `error`, causes the error I get when running the test

![image](https://github.com/user-attachments/assets/cc5bc8d5-9a10-4952-92b2-bf244049e4de)

How can I make it work? (Copied from [here](https://github.com/digitallyinduced/ihp/blob/7da31c2d02a7c8eb41a63ebc4d30bbf6aacbeafe/Test/Controller/CookieSpec.hs#L36-L42))

Reply:
That GHC error is unrelated to the error call. I just pushed a workaround via https://github.com/digitallyinduced/ihp/commit/66518dee183d71bc5a1e1df59c1e8019298a0c0c to master (appartently the INLINE triggers the bug in GHC, but chaging that to an optional inline via INLINABLE not triggers the bug). Can you sync with master and then check again?

Reply:
Thanks. I think now it hits that `error`

![image](https://github.com/user-attachments/assets/d1ae8e69-becd-4a72-aac7-c3e456172cdc)

Reply:
You can use `IHP.FrameworkConfig.withFrameworkConfig` to get a framework config object, like this:

```haskell
    let withFrameworkConfig = IHP.FrameworkConfig.withFrameworkConfig (pure ())
    describe "createTemporaryDownloadUrlFromPathWithExpiredAt" $ do
        it "returns baseUrl concatenated with objectPath when objectPath does not start with http:// or https://" $ do
            withFrameworkConfig \frameworkConfig -> do
                context <- createControllerContext frameworkConfig
                let ?context = context
                let objectPath = "static/test.txt"
                temporaryDownloadUrl <- createTemporaryDownloadUrlFromPath objectPath

                temporaryDownloadUrl.url `shouldBe` "http://localhost:8000/static/test.txt"
```

And then use the frameworkConfig passed to `createControllerContext` instead of the `error` call

Reply:
With this I get 

```
  ./IHP/Prelude.hs:92:17: 
  1) IHP.FileStorage.ControllerFunctions.storeFileWithOptions returns the objectPath without the baseUrl
       uncaught exception: ErrorCall
       Could not find FileStorage in config. Did you call initS3Storage from your Config.hs?
       CallStack (from HasCallStack):
         error, called at ./IHP/Prelude.hs:92:17 in main:IHP.Prelude
```

While trying to figure how to pass a config with the `initStaticDirStorage` (didn't figure yet), I found `aroundAll (withIHPApp WebApplication config) do`

Maybe we should use that? (If so, not sure how :sweat_smile: )

Reply:
I was able to inti the storage [here](https://github.com/digitallyinduced/ihp/pull/1990/commits/600d9feae61ff75157d6eec80735b894c1a8aff1)